[FIX] mass_mailing: reduce pad on template select

thiryr opened the pull request:
There was too much padding up and down from the tooltip.

Adds conditional display on the template menu
when there are no templates (t-for-each is not fully empty)

task #2984478

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

awa-odoo commented on addons/mass_mailing/static/src/xml/mass_mailing.xml:
```diff
@@ -8,10 +8,10 @@
             Click on the ⭐ next to the subject to save this mailing as a <span t-esc="modelName"/> template
         </div>
-        <div class="d-inline-block w-100">
+        <div class="d-inline-block w-100" t-if="template">
```
Shouldn't it be templateS here? Or maybe I missed something.

thiryr replied:
Ouch, that's embarassing. Yeah, I'll check that it works WITH templates too before I ping you again..

Thanks for the review 😅

awa-odoo reviewed (commented):
Hello @thiryr 

Well this is a short one, so not much to say, but I did find something strange :eyes: 

Thanks & cheers!

awa-odoo commented on addons/mass_mailing/static/src/js/mass_mailing_html_field.js:
```diff
@@ -255,6 +255,7 @@ export class MassMailingHtmlField extends HtmlField {
         const $themeSelectorNew = $(qweb.render("mass_mailing.theme_selector_new", {
             themes: themesParams,
             templates: templatesParams,
+            validTemplates: templatesParams.filter(template => template.modelId === this.props.record.data.mailing_model_id[0]),
```
Hello @thiryr 

Do not remember seeing this the first time.

What does "validTemplates" imply exactly?
It seems like the xml loops on the "templates" key anyway, so I'm not sure I understand the distinction.

thiryr replied:
I didn't take into account templates existing but not being displayed (because not the same 'model')

The 'invalid' templates are hidden when ``_hideIrrelevantTemplates`` is called, not sure why we're not filtering templates directly already (maybe for dynamic updates without using states, since it's rendered manually?).

I need this otherwise if there exists ANY template, the div is rendered with a space and we have a 1-character-high gap.


I could probably add the logic for this inside ``_hideIrrelevantTemplates`` to keep things in one place though.

awa-odoo replied:
I see there is indeed some specific logic in `_hideIrrelevantTemplates`.

What we could do is keep the code as you had before, but add logic into `_hideIrrelevantTemplates` to hide this wrapper div when it shows the "o_mailing_template_message" (and invert the logic in the else case).

This would require to add a specific class to target that div, something like "o_mailing_template_preview_wrapper".

That way we keep it simple and integrate into the existing code.

Does that make sense?

PS: Sorry for the delayed answer, it's crazy times with the v16 """freeze""" ^^

thiryr replied:
^^ no problem, this is very far from my oldest PR anyway.

I moved the logic inside ``_hideIrrelevantTemplates`` as discussed.

thiryr commented on addons/mail/static/src/scss/emojis.scss:
```diff
@@ -45,6 +45,13 @@ $o-mail-emoji-height: 2rem;
+        min-width: 8em;
+        max-width: 90%;
+    }
+    display: inline-block;
```
The emoji picker button didn't want to inline without this for some reason but I feel like I might have missed something,

thiryr commented:
@awa-odoo hey, the scope of this was increased a bit. I updated the char_emoji widget to use the same emoji picker as they use in discuss.

Reworking their implementation ended up being pretty tough so I just took inspiration from composer.js and used the messaging service/mini-framework to instanciate a new 'model' which in turn instanciates the popover via the... interesting mechanism that is already in place.

tldr; I used the discuss mini-framework and added a couple more 'if' statements in popover_view.js  Could you check this out again real quick before I rebase?

Thanks in advance.

std-odoo commented on addons/mass_mailing/static/src/js/mass_mailing_html_field.js:
```diff
@@ -424,9 +424,11 @@ export class MassMailingHtmlField extends HtmlField {
         // Hide or show the help message if some templates are visible
```
Can you update this comment ? (so we are 100% precise)

thiryr replied:
Sure :)
Made it "Hide or show the help message and preview wrapper based on whether there are any relevant templates"

std-odoo reviewed (commented):
@thiryr Hi :) 

One small comment, otherwise LGTM :)

std-odoo reviewed (approved):
LGTM :)

std-odoo commented:
@robodoo r+
